unstructured/ingest/connector/sql/connector.py

```diff
@@ -0,0 +1,216 @@
+            data["metadata"]["data_source"]["date_created"] = datetime.fromisoformat(date_created)
+
+        if date_modified := data.get("metadata", {}).get("data_source", {}).get("date_modified"):
+            data["metadata"]["data_source"]["date_modified"] = datetime.fromisoformat(date_modified)
+
+        if date_processed := data.get("metadata", {}).get("data_source", {}).get("date_processed"):
+            data["metadata"]["data_source"]["date_processed"] = datetime.fromisoformat(
+                date_processed
+            )
+
+        if last_modified := data.get("metadata", {}).get("last_modified", {}):
+            data["metadata"]["last_modified"] = datetime.fromisoformat(last_modified)
+
+        # String casting
+        if version := data.get("metadata", {}).get("data_source", {}).get("version"):
+            data["metadata"]["data_source"]["version"] = str(version)
+
+        if page_number := data.get("metadata", {}).get("page_number"):
+            data["metadata"]["page_number"] = str(page_number)
+
+        if regex_metadata := data.get("metadata", {}).get("regex_metadata"):
+            data["metadata"]["regex_metadata"] = str(json.dumps(regex_metadata))
+
+        if data.get("metadata", {}).get("data_source", None):
+            data.update(data.get("metadata", {}).pop("data_source", None))
+        if data.get("metadata", {}).get("coordinates", None):
+            data.update(data.get("metadata", {}).pop("coordinates", None))
+        if data.get("metadata", {}):
+            data.update(data.pop("metadata", None))
+
+        return data
+
+    def _resolve_mode(self, schema_helper: DatabaseSchema) -> t.Optional[dict]:
+        schema_exists = schema_helper.check_schema_exists()
+        if (
+            self.write_config.mode == "error" or self.write_config.mode == "ignore"
+        ) and schema_exists:
+            raise ValueError(
+                f"There's already an elements schema ({ELEMENTS_TABLE_NAME}) "
+                f"at {self.connector_config.db_type}"
+            )
+        if self.write_config.mode == "overwrite" and schema_exists:
+            schema_helper.clear_schema()
+
+    # @DestinationConnectionError.wrap
+    def write_dict(self, *args, json_list: t.List[t.Dict[str, t.Any]], **kwargs) -> None:
+        logger.info(
+            f"writing {len(json_list)} objects to database {self.connector_config.database} "
+            f"at {self.connector_config.host}"
+        )
+
+        with self.client as conn:
+            schema_helper = DatabaseSchema(
+                conn=conn,
+                db_type=self.connector_config.db_type,
+            )
+
+            self._resolve_mode(schema_helper)
+            for e in json_list:
+                elem = self.conform_dict(e)
+
+                schema_helper.insert(
+                    ELEMENTS_TABLE_NAME,
+                    elem,
+                )
+
+            conn.commit()
```

Comment:
Should there be a limit on how many insert statements we add in for each `commit`?  Not sure if the sdk batches them up for us or not.